### PR TITLE
8280166: Extend java/lang/instrument/GetObjectSizeIntrinsicsTest.java test cases

### PR DESCRIPTION
**Pull request by shipilev**
While working on JDK-8280003, I noticed that java/lang/instrument/GetObjectSizeIntrinsicsTest.java does not test arrays with more than 1-byte size elements, and no large arrays (past 4G limit) are tested either. It would be better to add those test cases. 

Additional testing:
 - [x] Linux x86_64 fastdebug, affected test still passes
 - [x] Linux x86_32 fastdebug, affected test still passes
 - [x] Linux AArch64 fastdebug, affected test still passes
 - [x] Linux PPC64 fastdebug, affected test still passes

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280166](https://bugs.openjdk.java.net/browse/JDK-8280166): Extend java/lang/instrument/GetObjectSizeIntrinsicsTest.java test cases

### Reviewers
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to 29e6fd24d4b31e308018dc72f0e408c3b548e42b
 * [Leonid Mesnik](https://openjdk.java.net/census#lmesnik) (@lmesnik - **Reviewer**) ⚠️ Review applies to dd9ce04964f0219876f7f943ddd9c20a35f31d67

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7132/head:pull/7132` \
`$ git checkout pull/7132`

Update a local copy of the PR: \
`$ git checkout pull/7132` \
`$ git pull https://git.openjdk.java.net/jdk pull/7132/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7132`

View PR using the GUI difftool: \
`$ git pr show -t 7132`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7132.diff">https://git.openjdk.java.net/jdk/pull/7132.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java [hunk 1]
```diff
@@ -264,25 +264,66 @@
...
+    final String mode;
+
+    public GetObjectSizeIntrinsicsTest(String name, String mode) {
         super(name);
+        this.mode = mode;
     }
 
     public static void main(String[] args)throws Throwable {
-        new GetObjectSizeIntrinsicsTest(args[0]).runTest();
+        new GetObjectSizeIntrinsicsTest(args[0], (args.length >= 2 ? args[1] : "")).runTest();
```
Comment (plummercj) [inline test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java:326]: Shouldn't this be `args.length == 2`?

Comment (shipilev) [reply test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java:326]: `>= 2` is more future-proof, I think, as it checks that _at least_ two arguments exists, which allows to read `args[1]`. Accidental addition of third argument would not silently break the test.

**Review by sspitsyn — APPROVED** [review]
Looks okay to me.
Thanks,
Serguei

File: test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java (hunk 1 above)
Comment (shipilev) [reply test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java:326]: @plummercj, are you good with this explanation?

**Review by lmesnik — APPROVED** [review]
Please update copyright years.

**Comment by shipilev** [comment]
> Please update copyright years.

Updated, thanks!

File: test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java (hunk 1 above)
Comment (shipilev) [reply test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java:326]: @plummercj This does not look like a review-blocking comment, so I am going to proceed with integration, if there are no other comments.

**Pull request closed without merging**